src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -8979,6 +8979,9 @@ class CL0<T>
 ", parseOptions: TestOptions.Regular8);
 
             c.VerifyDiagnostics(
+                // (22,22): warning CS8625: Cannot convert null literal to non-nullable reference.
```

Comment:
>// (22,22): warning CS8625: Cannot convert null literal to non-nullable reference. [](start = 16, length = 82)

We might want to report a different warning with message more specific to the unconstraint type parameter, or adjust the message for this warning to include information about special nature of an unconstraint type parameter. Something like: "Cannot convert null literal to non-nullable reference** or an unconstraint type parameter**."